typing/typemod.ml

```diff
@@ -2118,7 +2118,7 @@ let simplify_app_summary app_view = match app_view.arg with
   | Some arg ->
     let mty = arg.arg.mod_type in
     match arg.is_syntactic_unit , arg.path with
-    | true , _      -> Includemod.Error.Unit, mty
+    | true , _      -> Includemod.Error.Empty_struct, mty
```

Comment:
Naive question: what does `arg.is_syntactic_unit` mean now and why do we keep it? (Backwards-compatibility?) Is there a comment somewhere in the codebase that explains this?

Reply:
(Maybe it could/should be renamed `is_empty_struct`)

Reply:
This `is_syntactic_unit` field is the typechecker version of `Empty_struct`.

We kept it for backward compatibility reason to be able to coerce a `struct end` to an unit module in a context where an unit module was expected. This implicit conversion currently raises a warning to slowly nudge people away from relying on the generative interpretation of `F()`.